packages/experimental/Expander/windows/ReactNativeExpander/ExpanderView.cpp

```diff
@@ -0,0 +1,272 @@
+            }
+            else if (propertyName == "headerForegroundPressed") {
+                this->SetResourceColor(propertyValue, L"ExpanderHeaderForegroundPressed");
+            }
+            else if (propertyName == "headerBorderBrush") {
+                this->SetResourceColor(propertyValue, L"ExpanderHeaderBorderBrush");
+            }
+            else if (propertyName == "headerBorderPointerOverBrush") {
+                this->SetResourceColor(propertyValue, L"ExpanderHeaderBorderPointerOverBrush");
+            }
+            else if (propertyName == "headerBorderPressedBrush") {
+                this->SetResourceColor(propertyValue, L"ExpanderHeaderBorderPressedBrush");
+            }
+            else if (propertyName == "headerDisabledForeground") {
+                this->SetResourceColor(propertyValue, L"ExpanderHeaderDisabledForeground");
+            }
+            else if (propertyName == "headerDisabledBorderBrush") {
+                this->SetResourceColor(propertyValue, L"ExpanderHeaderDisabledBorderBrush");
+            }
+            else if (propertyName == "contentBackground") {
+                this->SetResourceColor(propertyValue, L"ExpanderContentBackground");
+            }
+            else if (propertyName == "contentBorderBrush") {
+                this->SetResourceColor(propertyValue, L"ExpanderContentBorderBrush");
+            }
+            else if (propertyName == "chevronBackground") {
+                this->SetResourceColor(propertyValue, L"ExpanderChevronBackground");
+            }
+            else if (propertyName == "chevronForeground") {
+                this->SetResourceColor(propertyValue, L"ExpanderChevronForeground");
+            }
+            else if (propertyName == "chevronPointerOverBackground") {
+                this->SetResourceColor(propertyValue, L"ExpanderChevronPointerOverBackground");
+            }
+            else if (propertyName == "chevronPointerOverForeground") {
+                this->SetResourceColor(propertyValue, L"ExpanderChevronPointerOverForeground");
+            }
+            else if (propertyName == "chevronPressedBackground") {
+                this->SetResourceColor(propertyValue, L"ExpanderChevronPressedBackground");
+            }
+            else if (propertyName == "chevronPressedForeground") {
+                this->SetResourceColor(propertyValue, L"ExpanderChevronPressedForeground");
+            }
+            else if (propertyName == "chevronBorderThickness") {
+                auto resDict = expander.Resources();
+                if (!propertyValue.IsNull()) {
+                    auto thickness = xaml::ThicknessHelper::FromUniformLength(propertyValue.AsDouble());
+                    resDict.Insert(winrt::box_value(L"ExpanderChevronBorderThickness"), winrt::box_value(thickness));
+                } else {
+                    resDict.Remove(winrt::box_value(L"ExpanderChevronBorderThickness"));
```

Comment:
can you clarify if the resource dictionary already has a value for this key upon creation, and we are Inserting into it to override the value, or if it's empty and we're inserting to set a value?

Does this work with fast refresh? IIRC ResourceDictionary is quirky in that it doesn't re-evaluate at runtime once an element is fully created and rendered

Reply:
The resource dictionary doesn't have a value for ExpanderChevronBorderThickness, and we are inserting to set a value. It is working with fast refresh.

If the resource dictionary doesn't have a key/value, is it pulled from [here](https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/Expander/Expander_themeresources.xaml)? Just curious how the relationship between the two works